public/app/plugins/datasource/jaeger/graphTransform.ts

```diff
@@ -0,0 +1,181 @@
+import { DataFrame, FieldType, MutableDataFrame } from '@grafana/data';
+import { NodeGraphDataFrameFieldNames as Fields } from '@grafana/ui';
+import { Span, TraceResponse } from './types';
+
+interface Node {
+  [Fields.id]: string;
+  [Fields.title]: string;
+  [Fields.subTitle]: string;
+  [Fields.mainStat]: string;
+  [Fields.secondaryStat]: string;
+  [Fields.color]: number;
+}
+
+interface Edge {
+  [Fields.id]: string;
+  [Fields.target]: string;
+  [Fields.source]: string;
+}
+
+export function createGraphFrames(data: TraceResponse): DataFrame[] {
+  const { nodes, edges } = convertTraceToGraph(data);
+
+  const nodesFrame = new MutableDataFrame({
+    fields: [
+      { name: Fields.id, type: FieldType.string },
+      { name: Fields.title, type: FieldType.string },
+      { name: Fields.subTitle, type: FieldType.string },
+      { name: Fields.mainStat, type: FieldType.string },
+      { name: Fields.secondaryStat, type: FieldType.string },
+      { name: Fields.color, type: FieldType.number, config: { color: { mode: 'continuous-GrYlRd' } } },
```

Comment:
So I wonder if this makes sense. Basically I am saying that with the node data you can also send a color column to say what color should the node have. But also you can specify a color mode and then use it's numeric value to get final color. This means I had to change the type of field.color.mode because the enum type does not contain all the modes from fieldColorModeRegistry and as that registry is dynamic it probably can't. I assume this makes sense from how fields work and should be easy to do overrides later on in panel for example (so this is sort of default mainly for Explore set by the data source) @torkelo @dprokop

Reply:
This is ok, but won't it make more sense to attach the color to the mainStat field?

Reply:
@torkelo it's not actually the same value. The color coding basically adds a third stat. Not sure how much sense that makes but we agreed with Tempo team that we would try it in ops to see how it works.

At the same time the mainStat is also not usable for this, because it is fancy string. You can send just a number with unit there but right now I also add a percentage after the number (to mimic what data Jaeger UI shows) and that's getting a bit complicated for the default field metadata.

Any feedback on this is welcome though.

Reply:
ah, ok

Reply:
so the color field actually contains a third stat values? Should not be named thirdStat then? if it's named color I would expect color string values

Reply:
@torkelo Yeah not sure about the naming. It actually can be a simple color string there is a code for that and could make sense for color coding based on the service name actually, ie spans from same service has same color it the trace view while the service colors are just randomized. So it could make sense to just assign the same static colors in both trace view and node graph. That's the reason the column name is just `color` and having numeric value with defined color scale is idea on top of that. As I said not sure what makes more sense for color coding whether the service name sync or the stat (self time as a percentage of whole trace). 

If the column was named `thirdStat` then having a static color name there would be more confusing. So I am on the fence cause I am not sure which color coding makes more sense and even if decide for a single one it seems to me like giving the option for the datasource to actually define it as it wants could be usefull.